subsys/bluetooth/host/gatt.c

```diff
@@ -4597,6 +4614,10 @@ int bt_gatt_unsubscribe(struct bt_conn *conn,
 		return -EINVAL;
 	}
 
+	if (gatt_sub_has_no_subscriptions(sub)) {
+		gatt_sub_free(sub);
+	}
+
 	if (has_subscription) {
```

Comment:
`gatt_sub_has_no_subscriptions` and `has_subscription` has very similar names, but are implemented different. 

Do they share the same semantics? i.e. if `has_subscription == false` is `gatt_sub_has_no_subscriptions` always true, or vice versa?